plugins/optimization-detective/storage/class-od-rest-url-metrics-store-endpoint.php

```diff
@@ -0,0 +1,347 @@
+<?php
+/**
+ * REST API integration for the plugin: OD_REST_URL_Metrics_Store_Endpoint.
+ *
+ * @package optimization-detective
+ * @since 0.1.0
+ */
+
+// @codeCoverageIgnoreStart
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+// @codeCoverageIgnoreEnd
+
+/**
+ * OD_REST_URL_Metrics_Store_Endpoint class
+ *
+ * @since n.e.x.t
+ * @access private
+ */
+final class OD_REST_URL_Metrics_Store_Endpoint {
+
+	/**
+	 * Namespace for the REST API endpoint.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	const REST_API_NAMESPACE = 'optimization-detective/v1';
+
+	/**
+	 * Route for storing a URL Metric.
+	 *
+	 * Note the `:store` art of the endpoint follows Google's guidance in AIP-136 for the use of the POST method in a way
+	 * that does not strictly follow the standard usage. Namely, submitting a POST request to this endpoint will either
+	 * create a new `od_url_metrics` post, or it will update an existing post if one already exists for the provided slug.
+	 *
+	 * @since n.e.x.t
+	 * @link https://google.aip.dev/136
+	 * @var string
+	 */
+	const REST_API_ROUTE = '/url-metrics:store';
```

Comment:
Likewise should this just be:

```suggestion
	const ROUTE = '/url-metrics:store';
```

Otherwise it is redundant.

Reply:
Resolved with - https://github.com/WordPress/performance/pull/1865#discussion_r1991808494